### ISSUE TITLE
VS2015.1 RC: Crash when inline renaming model type

### ISSUE THREAD

**Issue by Suchiman** [dotnet/roslyn#6821]
Stacktrace obtained from attaching VS to VS:
```
System.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.OpenTextBufferManager.SelectionTracking.<>c__DisplayClass5_0.<.ctor>b__0(KeyValuePair`2 kvp)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.OpenTextBufferManager.SelectionTracking..ctor(OpenTextBufferManager openTextBufferManager)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.OpenTextBufferManager.ApplyConflictResolutionEdits(IInlineRenameReplacementInfo conflictResolution, IEnumerable`1 documents, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.ApplyReplacements(IInlineRenameReplacementInfo replacementInfo, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.<>c__DisplayClass63_0.<UpdateConflictResolutionTask>b__1(Task`1 t)
   at Roslyn.Utilities.TaskExtensions.<>c__DisplayClass4_0`1.<SafeContinueWith>b__0(Task antecedent)
   at Roslyn.Utilities.TaskExtensions.<>c__DisplayClass1_0.<SafeContinueWith>b__0(Task antecedent)
   at Roslyn.Utilities.TaskExtensions.<>c__DisplayClass5_0`1.<SafeContinueWith>b__0(Task t)
```

I have two files, `Models.cs` and `EmailTemplate.cshtml` (latter using RazorEngine as email templating engine) in a web Project. The `EmailTemplate.cshtml` has a `@model SampleType` directive. When both files are opened and i try to rename `SampleType` inside `Models.cs`, VS will Crash. When i close `EmailTemplate.cshtml` prior to starting the inline rename, VS won't Crash.

**Comment by dpoeschl** [comment]
@Suchiman Thanks for the detailed report. We are able to reproduce this issue locally. 

Strangely, it doesn't reproduce every time -- sometimes the rename works as expected. The investigation is ongoing.

**Comment by wstaelens** [comment]
Model, Controller (that uses the model) and a View (that uses the model).
All 3 files open and F2 on Model in the Controller == crash.
Close the View and do F2 on the Model in the Controller == ok.

**Comment by pdelvo** [comment]
This also happens when renaming a property in a tag helper (that is used in a razor file).

**Comment by brettfo** [comment]
Note, this is also being tracked by internal bug 182787.

**Comment by dpoeschl** [comment]
Passing off to @brettfo, who has been driving this fix.

**Comment by Pilchie** [comment]
Fixed (in stabilization) by #9208.
